pkg/cloud/services/compute/instance.go

```diff
@@ -568,36 +567,38 @@ func (s *Service) AssociateFloatingIP(instanceID, floatingIP string) error {
 }
 
 func (s *Service) InstanceDelete(machine *clusterv1.Machine, openStackMachine *infrav1.OpenStackMachine) error {
-	if machine.Spec.ProviderID == nil {
+	if openStackMachine == nil {
 		// nothing to do
 		return nil
 	}
 
-	parsed, err := noderefutil.NewProviderID(*machine.Spec.ProviderID)
+	instance, err := s.InstanceExists(openStackMachine.Name)
```

Comment:
This whole block looks like the same as in bastion.go. Can we move this in a func and reuse it?

Reply:
I think you can use `object runtime.Object`